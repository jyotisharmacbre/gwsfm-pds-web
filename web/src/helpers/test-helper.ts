import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
