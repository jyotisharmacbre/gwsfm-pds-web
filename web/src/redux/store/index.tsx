import { applyMiddleware, createStore } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import allReducer from '../reducers';

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, allReducer);

const middleware = applyMiddleware(thunk);
const store = createStore(persistedReducer, middleware);
const persistor = persistStore(store);
export { store, persistor };
