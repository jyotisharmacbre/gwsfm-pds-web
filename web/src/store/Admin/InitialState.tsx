import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
