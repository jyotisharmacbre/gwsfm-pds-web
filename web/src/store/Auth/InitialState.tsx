import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
