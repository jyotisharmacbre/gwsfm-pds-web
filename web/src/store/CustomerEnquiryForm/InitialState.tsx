import { IProjectDetailState } from './Types/IProjectDetailState';
import EventType from '../../enums/EventType';
import Notify from '../../enums/Notify';

export const initialState: IProjectDetailState = {
    form: {
        projectId: '',
        name: '',
        contractorId: '',
        companyId: '',
        headOfProject: '',
        projectOwner: '',
        projectManager: '',
        pmHasExperience: false,
        scope: '',
        cnNumber: '',
        status: 1,
        engagementId: undefined,
        countryId: 0,
        currencyId: 0,
        probabilityOfWinning: '',
        approxValue: '',
        contractTypeId: '',
        cdmNotifiable: false,
        firstAssetWorkedOn: undefined,
        secondAssetWorkedOn: undefined,
        thirdAssetWorkedOn: undefined,
        comment: '',
        otherCompanyName: '',
        otherContractName: '',
        divisionId: '',
        otherDivision: '',
        businessUnitId: '',
        otherBusinessUnit: '',
        weightedTCV: undefined,
        soldMargin: undefined,
        otherEngagementType: '',
        otherFirstAssetWorkedOn: '',
        otherSecondAssetWorkedOn: '',
        otherThirdAssetWorkedOn: ''
    },
    enquiryOverview: {
        projectName: '',
        contractorId: '',
        headOfProject: '',
        projectManager: '',
        scope: '',
        cnNumber: '',
        otherContractName: ''
    },
    error: null,
    loading: false,
    notify: Notify.none,
    event: EventType.none,
    enquiryOverviewError: null
};