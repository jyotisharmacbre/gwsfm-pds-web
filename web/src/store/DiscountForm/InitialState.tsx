import { IDiscountState } from './Types/IDiscountState';
import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';
import IDiscountSubContractor from './Types/IDiscountSubContractor';

export const newSubContractorDiscount: IDiscountSubContractor = {
  subContractorDiscountId: '',
  projectId: '',
  supplierName: '',
  supplierState: '',
  supplierTotalDiscount: undefined,
  supplierComments: '',
}

export const initialState: IDiscountState = {
  form: {
    projectId: '',
    clientDiscount: {
      discountId: '',
      projectId: '',
      discount: undefined,
      clientState: '',
      discountType: 1,
      clientComments: ''
    },
    subContractorDiscounts: [{ ...newSubContractorDiscount }],
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};