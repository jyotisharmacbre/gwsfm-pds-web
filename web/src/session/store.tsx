import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger'
import rootReducer from './reducers/RootReducer';

const middlewares: any[] = [];

middlewares.push(thunk);

if (process.env.NODE_ENV === `development`) {

  middlewares.push(logger);
}

export default function configureStore(initialState = {}) {
  return createStore(
    rootReducer,
    applyMiddleware(...middlewares)
  );
}