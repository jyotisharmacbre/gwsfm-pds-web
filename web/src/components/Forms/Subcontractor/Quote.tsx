import React from 'react';
import { Field, reduxForm, InjectedFormProps, FormSection } from 'redux-form';
import IReactIntl from '../../../Translations/IReactIntl';
import { IQuote } from '../../../store/SubContractor/Types/IQuote';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import PdsFormButton from '../../PdsFormHandlers/PdsFormButton';
import PdsFormTypeAhead from '../../PdsFormHandlers/PdsFormTypeAhead';
import { FormattedMessage, injectIntl } from 'react-intl';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import ValidatedNumericInput from '../../NumericInput';
import { restrictMinus, restrictMinusAndAllowDecimal } from '../../../helpers/utility-helper';
import { newQuote } from '../../../store/SubContractor/InitialState';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlusCircle, faTrash } from '@fortawesome/free-solid-svg-icons';
import { confirmAlert } from '../../Popup/CustomModalPopup';

interface IProps {
    fields: any;
    currencySymbol: string;
    intl: any;
}

const Quote: React.FC<IProps> = props => {
    const deleteQuote = (index) => {
        debugger;
        props.fields.remove(index);
    }
    return (
        <div className="row" data-test="sub-contractor-quote-form">
            <div className="col-lg-12">
                <div className="card_outer_wrap quote_wrap">
                    <div className="row">
                        {props.fields.map((member, index) => (
                            <div key={index} className="col-lg-4 pl-md-2 pr-md-2 mb-2 mt-2" data-test="sub-contractor-quote-member">
                                <div className="card_wrap">
                                    <div className="card">
                                        <h6 className="title"><FormattedMessage id='TITLE_QUOTE'></FormattedMessage> {index + 1}</h6>
                                        {props.fields.length > 3 ? (
                                            <button
                                                data-test="deletequote"
                                                className="delete_text"
                                                onClick={() =>
                                                    confirmAlert({
                                                        intl: props.intl,
                                                        titleKey: 'TITLE_CONFIRMATION',
                                                        contentKey: 'MESSAGE_SUBCONTRACTOR_DELETE_QUOTE',
                                                        handleConfirm: () => deleteQuote(index) 
                                                    })}
                                            >
                                                <FontAwesomeIcon className="" icon={faTrash} />
                                            </button>
                                        ) : null}
                                        <Field
                                            name={`${member}.supplierName`}
                                            data-test="supplierName"
                                            type="text"
                                            component={PdsFormInput}
                                            validate={[
                                                Validate.maxLength(1000)
                                            ]}
                                            labelKey="LABEL_ACTIVITY_NAME"
                                            placeholderKey="PLACEHOLDER_SUPPLIER_NAME"
                                        />
                                        <Field
                                            name={`${member}.quoteValue`}
                                            type="text"
                                            component={PdsFormInput}
                                            className="pl-20"
                                            validate={[
                                                Validate.maxLength(1000),
                                                onlyNumber
                                            ]}
                                            currency={props.currencySymbol}
                                            divPosition="relative"
                                            labelKey="LABEL_QUOTE_VALUE"
                                            placeholderKey=""
                                            normalize={restrictMinusAndAllowDecimal}
                                        />
                                    </div>
                                </div>
                            </div>
                        ))}
                        <div className="newActiv_btn">
                            <button
                                data-test="addQuote"
                                name="addQuote"
                                type="button"
                                className="active"
                                onClick={() => {
                                    props.fields.length > 19 ?
                                        confirmAlert({
                                            intl: props.intl,
                                            contentKey: 'MESSAGE_SUBCONTRACTOR_ADD_MAX_QUOTE',
                                            showFooter: false
                                        })
                                        :
                                        props.fields.push({ ...newQuote });
                                }}
                            >
                                <FontAwesomeIcon className="" icon={faPlusCircle} />
                                <FormattedMessage id="BUTTON_NEW_QUOTE" />
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        )
}

export default Quote; 
