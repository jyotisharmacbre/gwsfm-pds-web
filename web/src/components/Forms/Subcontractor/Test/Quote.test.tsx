import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import Quote from '../Quote';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import   * as connectedIntlProvider from './../../../../Translations/connectedIntlProvider';
import {newQuote,initialState} from '../../../../store/SubContractor/InitialState';
import {findByTestAtrr} from '../../../../helpers/test-helper';

describe('Sub Contractor Quote Form component tests', () => {
  let wrapper: any;
  let quotes =[{...newQuote},{...newQuote},{...newQuote}]
  beforeEach(() => {
    wrapper = shallow(
      <Quote fields={initialState.form.activities[0].quotes}/>
    );
  });

it('should match the snapshot', () => { 
    expect(wrapper).toMatchSnapshot();
  });
  it('should defines the component', () => {
    expect(wrapper).toBeDefined();
  });
  it('renders the SubContractorActivityForm component with no errors', () => {
    
    expect(findByTestAtrr(wrapper,'sub-contractor-quote-form').length).toEqual(1);
    expect(findByTestAtrr(wrapper,'sub-contractor-quote-member').length).toEqual(3);
  });

  
});
