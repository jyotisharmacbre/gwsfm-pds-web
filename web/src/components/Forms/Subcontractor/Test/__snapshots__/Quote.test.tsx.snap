// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Contractor Quote Form component tests should match the snapshot 1`] = `ShallowWrapper {}`;
